server/services/event.js

```diff
@@ -299,16 +299,31 @@ module.exports = class EventService {
             playerIdAttackers: attackers.map(p => p._id),
             starId: star._id,
             starName: star.name,
-            combatResult,
             captureResult
         };
 
         for (let defender of defenders) {
-            await this.createPlayerEvent(gameId, gameTick, defender._id, this.EVENT_TYPES.PLAYER_COMBAT_STAR, data);
+            let defenderCombatResult = combatResult
+            defenderCombatResult.carriers = combatResult.carriers.map(c => {
+                if (c.specialist && c.specialist.name === 'Scrambler' && !defender._id.equals(c.ownedByPlayerId) && c.after !== 0) {
+                    return { ...c, before: '???', after: '???' }
+                }
+                return c
+            })
+            await this.createPlayerEvent(gameId, gameTick, defender._id, this.EVENT_TYPES.PLAYER_COMBAT_STAR, { ...data, combatResult: defenderCombatResult });
         }
 
         for (let attacker of attackers) {
-            await this.createPlayerEvent(gameId, gameTick, attacker._id, this.EVENT_TYPES.PLAYER_COMBAT_STAR, data);
+            let attackerCombatResult = combatResult
+            let bool = attackerCombatResult.star.specialist && attackerCombatResult.star.specialist.name === 'Scrambler' && (captureResult !== null)
+            attackerCombatResult.star = { ...combatResult.star, before: bool ? '???' : combatResult.star.before, after: bool ? '???' : combatResult.star.after }
+            attackerCombatResult.carriers = combatResult.carriers.map(c => {
+                if (c.specialist && c.specialist.name === 'Scrambler' && !attacker._id.equals(c.ownedByPlayerId) && c.after !== 0) {
```

Comment:
When calculating whether a specialist is a scrambler, use `c.specialist.modifiers.special.hideCarrierShips` instead. You may need to null check `modifiers` and `special`.

Reply:
c.specialist?.modifiers?.special?.hideCarrierShips